Dataset.py

```diff
@@ -606,6 +645,10 @@ def get_data(self, seq_idx, key):
     :param str key: data-key, e.g. "data" or "classes"
     :rtype: numpy.ndarray
     :returns features or targets: format 2d (time,feature) (float)
+
+    TODO: the dependencies between the different get_X_data functions is not clear,
+      e.g. HDFDatasets calls this function from the other two
```

Comment:
They are there for historical reasons.
In an optimal world, you would only overwrite `get_data`, and also only use `get_data`.